### PR TITLE
luci-app-watchcat: add support for run_script mode

### PR DESCRIPTION
**Pull request by jtkohl**
This adds support for run_script mode, which I propose adding in https://github.com/openwrt/packages/pull/18593

I also added support for choosing the address family for the pings

@feckert @nickberry17 

I apologize, I don't know anything about the translated messages and how to deal with those in this package.  Can you point me at some documentation about that?

### CODE REVIEW

File: applications/luci-app-watchcat/htdocs/luci-static/resources/view/watchcat.js [hunk 1]
```diff
@@ -22,29 +22,49 @@ return view.extend({
...
 				<br /><br />The default unit is seconds, without a suffix, but you can use the \
 				suffix <b>m</b> for minutes, <b>h</b> for hours or <b>d</b> \
 				for days. <br /><br />Examples:<ul><li>10 seconds would be: <b>10</b> or <b>10s</b></li><li>5 minutes would be: <b>5m</b></li><li> \
 				1 hour would be: <b>1h</b></li><li>1 week would be: <b>7d</b></li><ul>"));
 		o.default = '6h';
 
-		o = s.taboption('general', form.Value, 'pinghosts', _('Host To Check'), _(`IPv4 address or hostname to ping.`));
-		o.datatype = 'host(1)';
+		o = s.taboption('general', form.Value, 'pinghosts', _('Host To Check'), _(`IP address or hostname to ping.`));
+		o.datatype = 'host';
```
Comment (nickberry17) [inline applications/luci-app-watchcat/htdocs/luci-static/resources/view/watchcat.js:53]: what was wrong with `host(1)`?

**Review by nickberry17 — COMMENTED** [review]
Looks good, just want to clarify about the datatype

File: applications/luci-app-watchcat/htdocs/luci-static/resources/view/watchcat.js (hunk 1 above)
Comment (jtkohl) [reply applications/luci-app-watchcat/htdocs/luci-static/resources/view/watchcat.js:53]: That didn't allow IPv6 literals, such as `2001::1`

**Comment by jtkohl** [comment]
> Looks good, just want to clarify about the datatype

Thanks, see above.  How about translation--is there something I need to do there?

**Comment by kofec** [comment]
@nickberry17 Hi, The main script was merged. Can we also merge these changes ?

**Comment by nickberry17** [comment]
LGTM @feckert

**Comment by jtkohl** [comment]
> LGTM @feckert

When might this get into master (and also if possible 22.03, which has the underlying changes in watchcat package)?

**Comment by feckert** [comment]
I must overlooked it!
Thanks merged

**Comment by feckert** [comment]
Also pushed to openwrt-22.03 with https://github.com/openwrt/luci/commit/adeb2baa3a14d8a0b4fdcc1cad067db380fb1306 and https://github.com/openwrt/luci/commit/010119fb5ab56f3e8c6d9d51f014771c1ef08e72

**Pull request merged**